app/javascript/src/components/PortEditor.tsx

```diff
@@ -292,7 +292,7 @@ class PortEditor extends React.Component<PortEditorProps & PortEditorStateProps,
 
   setShowIfStatValue = (index: number, e: React.FocusEvent<HTMLInputElement>) => {
     let newShowIfStats = clone(this.state.thisPortMeta.showIfStats || [])
-    newShowIfStats[index].value = e.target.value
+    newShowIfStats[index].value = parseInt(e.target.value, 0)
```

Comment:
We were assigning strings to a number field, which was probably accidentally working because in JS '1' + 1 is 2, but could have broken in unexpected ways down the line.